src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -26,6 +26,7 @@ public unsafe static uint[] Square(uint[] value)
 
         // Mutable for unit testing...
         private static int SquareThreshold = 32;
+        private static int AllocationThreshold = 4096;
```

Comment:
This is bit length, right?  Just curious how you arrived at this value for the threshold?